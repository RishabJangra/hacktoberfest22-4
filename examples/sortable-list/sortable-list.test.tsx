import React from 'react';
import renderer from 'react-test-renderer';
import { SortableListScreen } from './SortableList';

jest.mock('@react-navigation/native', () => ({
    useNavigation: (): any => ({ openDrawer: jest.fn(() => true) }),
}));

describe('Sortable List Tests', () => {
    it('Screen Renders', () => {
        const tree = renderer.create(<SortableListScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
