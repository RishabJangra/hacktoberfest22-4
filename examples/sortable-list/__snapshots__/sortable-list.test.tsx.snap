// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sortable List Tests renders the screen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    accessible={true}
    emulateUnlessSupported={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#6200ee",
        "elevation": 0,
        "height": 76,
        "shadowColor": "rgba(0, 0, 0, 0.3)",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 2,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "paddingBottom": 16,
          "paddingHorizontal": 16,
          "paddingTop": 16,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 40,
            "margin": -8,
            "marginRight": 24,
            "opacity": 1,
            "padding": 8,
            "width": 40,
          }
        }
        testID="header-navigation"
      >
        <Text />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "flex-end",
            "marginRight": 40,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 0,
              "justifyContent": "center",
            }
          }
        >
          <Text
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "white",
                "fontFamily": "System",
                "fontSize": 20,
                "lineHeight": 20,
                "textAlign": "auto",
                "writingDirection": "ltr",
              }
            }
            testID="header-title"
          >
            Sortable List
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 56,
              "position": "absolute",
              "right": 8,
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": 40,
              "opacity": 1,
              "padding": 8,
              "width": 40,
            }
          }
          testID="header-action-item0"
        >
          <Text />
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
  <RCTScrollView
    data={
      Array [
        Object {
          "name": "George Washington",
          "value": 1789,
        },
        Object {
          "name": "John Adams",
          "value": 1796,
        },
        Object {
          "name": "Thomas Jefferson",
          "value": 1800,
        },
        Object {
          "name": "James Madison",
          "value": 1808,
        },
        Object {
          "name": "James Monroe",
          "value": 1812,
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flexDirection": "row",
              "height": 72,
              "opacity": 1,
              "paddingRight": 16,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": undefined,
                  "height": "100%",
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 6,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "justifyContent": "center",
                  "marginLeft": 16,
                  "width": 40,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "paddingHorizontal": 16,
                },
                undefined,
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {},
                  Object {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    undefined,
                  ],
                ]
              }
            >
              George Washington
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
          </View>
          <Text>
            1789
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flexDirection": "row",
              "height": 72,
              "opacity": 1,
              "paddingRight": 16,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": undefined,
                  "height": "100%",
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 6,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "justifyContent": "center",
                  "marginLeft": 16,
                  "width": 40,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "paddingHorizontal": 16,
                },
                undefined,
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {},
                  Object {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    undefined,
                  ],
                ]
              }
            >
              John Adams
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
          </View>
          <Text>
            1796
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flexDirection": "row",
              "height": 72,
              "opacity": 1,
              "paddingRight": 16,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": undefined,
                  "height": "100%",
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 6,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "justifyContent": "center",
                  "marginLeft": 16,
                  "width": 40,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "paddingHorizontal": 16,
                },
                undefined,
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {},
                  Object {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Thomas Jefferson
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
          </View>
          <Text>
            1800
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flexDirection": "row",
              "height": 72,
              "opacity": 1,
              "paddingRight": 16,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": undefined,
                  "height": "100%",
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 6,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "justifyContent": "center",
                  "marginLeft": 16,
                  "width": 40,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "paddingHorizontal": 16,
                },
                undefined,
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {},
                  Object {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    undefined,
                  ],
                ]
              }
            >
              James Madison
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
          </View>
          <Text>
            1808
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "flexDirection": "row",
              "height": 72,
              "opacity": 1,
              "paddingRight": 16,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": undefined,
                  "height": "100%",
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": 6,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "justifyContent": "center",
                  "marginLeft": 16,
                  "width": 40,
                },
                undefined,
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "flex": 1,
                  "paddingHorizontal": 16,
                },
                undefined,
              ]
            }
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": "#000000",
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Object {},
                  Object {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                  Array [
                    Object {
                      "color": "#000000",
                    },
                    undefined,
                  ],
                ]
              }
            >
              James Monroe
            </Text>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  },
                  undefined,
                ]
              }
            />
          </View>
          <Text>
            1812
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
